### ISSUE TITLE
scripts with type different than "module" using dynamic import.

### ISSUE THREAD

**Issue by youknowriad** [guybedford/es-module-shims#156]
Hi there and thanks for your work on this polyfill.

I noticed that in Chrome, we can do `import( 'some module' )` in any script file, it doesn't matter if it's a `type="module"` or not. And after reading a bit here, it seems that the polyfill doesn't implement support for this. Is this a design choice or some kind of limitation?

Thanks.

**Comment by guybedford** [comment]
@youknowriad yes this polyfill doesn't touch or polyfill scripts so won't affect dynamic import in non modules.

This could be more clearly documented as a limitation of the approach certainly. If you'd like to suggest wording along these lines that could help.

**Comment by youknowriad** [comment]
Hi thanks for the quick reply.

To be honest, I'm very bad with wording suggestions :P so I'd prefer to leave that one for you and other contributors :) 

So no plans to consider this in the future right?

**Comment by guybedford** [comment]
Unless there's a really compelling use case for this I would rather avoid it just for predictability of the polyfilling as well. If we parse all scripts on the page that seems like it might be this project overstepping its boundaries to me possibly. There is also the problem of double execution since all code up to the dynamic import would execute twice.

If you need to use a shimmed dynamic import in scripts I would advise just manually referencing `importShim(...)` instead of `import(...)` and things should work out. It will still use the native loader as much as possible and avoid unnecessary work.

**Comment by youknowriad** [comment]
Happy to clarify my use-case more:

I'm working on trying to introduce import maps and ES modules to WordPress https://github.com/WordPress/gutenberg/pull/34172 

The scripts that could potentially make use of new ES modules dependencies are existing non module WP scripts. And these are part of the public API of WordPress that plugins and themes are relying on, on their own JS scripts. Meaning, It's not possible for us (WordPress Core) to rewrite these existing scripts as `type="module"` without introducing breaking changes for third-party WP plugins and themes. (WordPress has almost 0 breaking change policy)

> If you need to use a shimmed dynamic import in scripts I would advise just manually referencing importShim(...) instead of import(...) and things should work out. It will still use the native loader as much as possible and avoid unnecessary work.

Yes, this is the approach I'm considering for now, it has some drawbacks for us though, in the sense that in the future, that "shim" could get removed when dynamic import support is good enough (I guess we can still ship `const importShim = import;` at that point.

**Comment by guybedford** [comment]
Very interesting to hear and thanks for the background here.

Can you perhaps share a full code sample of the sort of code using dynamic import in scripts that you can come across here?
